src/developer-documentation/ibc/ibc.md

```diff
@@ -0,0 +1,999 @@
+        // /// This is the same structure as messages in `TxBody` from [ADR-020](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-020-protobuf-transaction-encoding.md)
+        // #[cfg(feature = "stargate")]
+        Stargate {
+            type_url: String,
+            value: Binary,
+        },
+        // #[cfg(feature = "stargate")]
+        Ibc(IbcMsg),
+        Wasm(WasmMsg),
+        // #[cfg(feature = "stargate")]
+        Gov(GovMsg),
+    }
+
+    pub struct SubMsg<T = Empty> {
+        /// An arbitrary ID chosen by the contract.
+        /// This is typically used to match `Reply`s in the `reply` entry point to the submessage.
+        pub id: u64,
+        pub msg: CosmosMsg<T>,
+        /// Gas limit measured in [Cosmos SDK gas](https://github.com/CosmWasm/cosmwasm/blob/main/docs/GAS.md).
+        pub gas_limit: Option<u64>,
+        pub reply_on: ReplyOn,
+    }
+
+    pub enum ReplyOn {
+        /// Always perform a callback after SubMsg is processed
+        Always,
+        /// Only callback if SubMsg returned an error, no callback on success case
+        Error,
+        /// Only callback if SubMsg was successful, no callback on error case
+        Success,
+        /// Never make a callback - this is like the original CosmosMsg semantics
+        Never,
+    }
+
+    pub enum QueryRequest<C> {
+        //Bank(BankQuery),
+        Custom(C),
+        // #[cfg(feature = "staking")]
+        Staking(StakingQuery),
+        /// A Stargate query is encoded the same way as abci_query, with path and protobuf encoded request data.
+        /// The format is defined in [ADR-21](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-021-protobuf-query-encoding.md).
+        /// The response is protobuf encoded data directly without a JSON response wrapper.
+        /// The caller is responsible for compiling the proper protobuf definitions for both requests and responses.
+        #[cfg(feature = "stargate")]
+        Stargate {
+            /// this is the fully qualified service path used for routing,
+            /// eg. custom/cosmos_sdk.x.bank.v1.Query/QueryBalance
+            path: String,
+            /// this is the expected protobuf message type (not any), binary encoded
+            data: Binary,
+        },
+        #[cfg(feature = "stargate")]
+        Ibc(IbcQuery),
+        Wasm(WasmQuery),
+    }
+
+    /// Defines the storage of your contract.
+    /// Add new fields to the below struct in order
+    /// to add new static storage fields to your contract.
+    #[ink(storage)]
+    pub struct Ics27demo {
+        /// config static string "config"
+        key_config: Vec<u8>,
+        ///pending_channel static string "pending"
+        key_pending_channel: Vec<u8>,
+        ///prefix_accounts static string "accounts"
+        prefix_accounts: Vec<u8>,
+        ///result static string "result"
+        result_prefix: Vec<u8>,
+    }
+
+    // pub enum ReflectExecuteMsg {
```

Comment:
It there a reason to keep commented blocks for documentation?

Reply:
no need